webapp/channels/src/packages/mattermost-redux/src/selectors/entities/scheduled_posts.ts

```diff
@@ -23,3 +23,21 @@ export function makeGetScheduledPostsByTeam(): (state: GlobalState, teamId: stri
         },
     );
 }
+
+export function getScheduledPostsByTeamCount(state: GlobalState, teamId: string, includeDirectChannels: boolean) {
+    let count = state.entities.scheduledPosts.byTeamId[teamId]?.length || 0;
+    if (includeDirectChannels) {
+        count += (state.entities.scheduledPosts.byTeamId.directChannels?.length || 0);
+    }
+
+    return count;
+}
+
+export function hasScheduledPostError(state: GlobalState, teamId: string) {
+    let hasError = (state.entities.scheduledPosts.byTeamId[teamId] || []).some((scheduledPost) => scheduledPost.error_code !== '');
+    if (!hasError) {
+        hasError = (state.entities.scheduledPosts.byTeamId.directChannels || []).some((scheduledPost) => scheduledPost.error_code !== '');
+    }
+
+    return hasError;
+}
```

Comment:
Selectors are run every time anything changes in the state anywhere. So, for example, if we receive a post through the websocket, the state will change, and any active selector will run. That is why selectors should be ultrafast.

If the list of scheduled posts grow, this could end up being annoying, since we would have to run over two lists.

I think this selector is only active in one place at a time, its complexity it's only O(n) and we don't expect the list of scheduled posts to get too high, so I am happy to leave it as is just to reduce code and state complexity. But if you feel any of my premises is wrong, please review this selector.

Reply:
Done. Refactored this by saving the error state of team in store directly. I compute it when I fetch all the scheduled posts, and then update it on receiving individual scheduled post. So now, checking for error is O(1) time.